### PR TITLE
feat: change detection filter

### PR DESCRIPTION
**Pull request by jobulcke**

### CODE REVIEW

File: ldi-core/change-detection-filter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        <artifactId>ldi-core</artifactId>
+        <version>2.6.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>change-detection-filter</artifactId>
+
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (Yalz) [inline ldi-core/change-detection-filter/pom.xml:17]: please remove these, they can only conflict with the parent :)

File: ldi-core/change-detection-filter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        <dependency>
+            <groupId>com.github.gwenn</groupId>
+            <artifactId>sqlite-dialect</artifactId>
+            <version>${sqlite-dialect.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>${postgresql.version}</version>
```
Comment (Yalz) [inline ldi-core/change-detection-filter/pom.xml:42]: test scope please

File: ldi-core/change-detection-filter/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/ChangeDetectionFilter.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	private void canonicalizeInputModel(Model model, OutputStream outputStream) {
+		final RdfDataset normalisedDataset = RdfNormalize.normalize(readDatasetFromJenaModel(model));
+		writeToOutputStream(normalisedDataset, outputStream);
+	}
+
+	private RdfDataset readDatasetFromJenaModel(Model model) {
+		final ByteArrayOutputStream receivedModelOutputStream = new ByteArrayOutputStream();
+		RDFWriter.source(model).lang(Lang.NQUADS).output(receivedModelOutputStream);
```
Comment (Yalz) [inline ldi-core/change-detection-filter/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/ChangeDetectionFilter.java:76]: I would put this NQUADS as a variable on top if you ever want to change

File: ldi-nifi/ldi-nifi-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/PersistenceProperties.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import static java.lang.Boolean.FALSE;
+import static java.lang.Boolean.TRUE;
+
+public class PersistenceProperties {
+	private PersistenceProperties() {
+	}
+
+	public static final PropertyDescriptor STATE_PERSISTENCE_STRATEGY = new PropertyDescriptor.Builder()
+			.name("STATE_PERSISTENCE_STRATEGY")
+			.displayName("How state is persisted (note that memory is volatile).")
```
Comment (Yalz) [inline ldi-nifi/ldi-nifi-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/PersistenceProperties.java:21]: very long name for display

File: ldi-nifi/ldi-nifi-processors/change-detection-processor/pom.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        <version>2.6.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>change-detection-processor</artifactId>
+
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (Yalz) [inline ldi-nifi/ldi-nifi-processors/change-detection-processor/pom.xml:18]: again, please leave these out

File: ldi-nifi/ldi-nifi-processors/change-detection-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/ChangeDetectionFilterRelationships.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldi.processors.config;
+
+import org.apache.nifi.processor.Relationship;
+
+public class ChangeDetectionFilterRelationships {
+	private ChangeDetectionFilterRelationships() {
+	}
+
+	public static final Relationship NEW_STATE_RECEIVED = new Relationship.Builder()
+			.name("new state received")
```
Comment (Yalz) [inline ldi-nifi/ldi-nifi-processors/change-detection-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/ChangeDetectionFilterRelationships.java:10]: Camelcase please

File: ldi-nifi/ldi-nifi-processors/change-detection-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/ChangeDetectionFilterRelationships.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+			.description("State member has changed and gone successfully through the filter")
+			.build();
+
+	public static final Relationship IGNORED = new Relationship.Builder()
+			.name("ignored")
```
Comment (Yalz) [inline ldi-nifi/ldi-nifi-processors/change-detection-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/config/ChangeDetectionFilterRelationships.java:15]: Camelcase please

File: ldi-orchestrator/ldio-connectors/ldio-change-detection-filter/pom.xml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        <version>2.6.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>ldio-change-detection-filter</artifactId>
+
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
```
Comment (Yalz) [inline ldi-orchestrator/ldio-connectors/ldio-change-detection-filter/pom.xml:18]: Please leave this out

File: ldi-orchestrator/ldio-connectors/ldio-repository-materialiser/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioRepositoryPipelineEventsListenerConfig.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+		return new LdioPipelineEventsListener.Builder<LdioRepositoryMaterialiser>()
+				.withStartBehavior(LdioRepositoryMaterialiser::start)
+				.withResumeBehavior(LdioRepositoryMaterialiser::start)
+				.withStopBehavior(LdioRepositoryMaterialiser::shutdown)
+				.withPauseBehavior(materialiser -> {
+					materialiser.sendToMaterialiser();
+					materialiser.shutdown();
+				})
+				.build();
+	}
```
Comment (Yalz) [inline ldi-orchestrator/ldio-connectors/ldio-repository-materialiser/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioRepositoryPipelineEventsListenerConfig.java:21]: although out of scope, can you explain why this was introduced?

**Review by Yalz — APPROVED** [review]

File: ldi-core/change-detection-filter/pom.xml (hunk 2 above)
Comment (jobulcke) [reply ldi-core/change-detection-filter/pom.xml:42]: Sorry, I thought this was managed by the dependencyManagment of the parent, but it's not

**Pull request merged**